src/branch.rs

```diff
@@ -28,6 +28,16 @@ impl<'repo> Branch<'repo> {
         Branch { inner: reference }
     }
 
+    /// Ensure the branch name is well-formed.
+    pub fn name_is_valid(name: &str) -> bool {
```

Comment:
it should be easy to write a unittest for this

Reply:
done, thanks!